### PR TITLE
Add support for LPC54114 & LPC54608 LPCXpresso boards

### PR DESCRIPTION
**Pull request by mmahadevan108**
Add support for LPC4322 interface
Add support for the LPC54114 target

### CODE REVIEW

**Comment by mmahadevan108** [comment]
@flit @c1728p9 @jasonNXP

File: source/daplink/RTX_Config.c [hunk 1]
```diff
@@ -90,7 +90,7 @@
 #elif defined(INTERFACE_SAM3U2C)
 #define OS_CLOCK    96000000
 #elif defined(INTERFACE_LPC4322)
-#define OS_CLOCK    204000000
+#define OS_CLOCK    96000000
```
Comment (c1728p9) [inline source/daplink/RTX_Config.c]: Could you keep the core clock at 204MHz? This will make debugging much faster.

Comment (mmahadevan108) [inline source/daplink/RTX_Config.c]: I checked with the LPC team internally, their recommendation was to slow the core clock down as the chip is not stable when running at this speed.

**Comment by mmahadevan108** [comment]
Thanks Russ, I have not tested on all platforms listed in the Acceptance Criteria document.

File: source/daplink/RTX_Config.c (hunk 1 above)
Comment (c1728p9) [inline source/daplink/RTX_Config.c]: How fast can the LPC4322 stably run?

**Comment by c1728p9** [comment]
Hi @mmahadevan108, running it on just one platform should be good

**Comment by mmahadevan108** [comment]
96MHz is the most stable frequency. We can look at raising the core frequency later. 

We would like to get a working implementation available to enable mbed-os work.

**Comment by mmahadevan108** [comment]
Updated pull request to move the reset code to a .c file.

**Comment by mmahadevan108** [comment]
@c1728p9 I was able to pass the DAPLink tests on my Windows 7 laptop.

**Comment by mmahadevan108** [comment]
Hi Russ, Anything else needed for this pull request.

**Comment by c1728p9** [comment]
I'll need to run testing before this can be merged. It would also be good to bring the speed back to 204MHz and to get USB working reliably, but those aren't blockers.

**Comment by mmahadevan108** [comment]
Thanks Russ. Boards are being shipped to Mac, I will request he give you one for your testing.

**Comment by mmahadevan108** [comment]
I have included support for LPC54608 as well in this branch. This is also using the LPC4322 interface chip.

**Comment by theotherjimmy** [comment]
@mmahadevan108 Could you change the PR title to reflect the extra target addition?

**Comment by mmahadevan108** [comment]
@c1728p9 @theotherjimmy Can this PR be integrated into the mainline?

**Comment by c1728p9** [comment]
Unfortunately CI is down right now. Once its up I can run testing and merge it.

**Comment by mmahadevan108** [comment]
@c1728p9 I noticed some instabilities with the UART driver for LPC4322. I update the buffer handling mechanism to use DAPLink circular buffer library. I leveraged the code from the LPC11U35 UART driver. Were you able to test this on the LPC54114 and LPC54608.

**Comment by c1728p9** [comment]
Hi @mmahadevan108, I have not had a chance to test yet

**Comment by mmahadevan108** [comment]
@c1728p9. Can this be integrated into the DAPLink code-base

File: source/daplink/RTX_Config.c (hunk 1 above)
Comment (c1728p9) [inline source/daplink/RTX_Config.c]: Can you provide details on the speed the lpc4322 can stably run for your configuration based on the voltage it is running at?

Comment (mmahadevan108) [inline source/daplink/RTX_Config.c]: 96MHz is the default frequency upon power up, the voltage is setup to stably run at this frequency.

Comment (c1728p9) [inline source/daplink/RTX_Config.c]: Older boards were using 204MHz. Is there a new board that is powering the LPC4322 with a lower voltage? If so can you provide any details on that and the voltage it operates at?

Comment (mmahadevan108) [inline source/daplink/RTX_Config.c]: I am not sure when this was tested. My experience with the current DAPLink code was it was not stable at the higher frequency. Hence I have slowed down to the default power up frequency.  I am checking internally for information on how to safely raise the core frequency.
Once I receive this, I will try it and submit a separate PR to raise the core frequency.

**Comment by c1728p9** [comment]
This PR has conflicts with master and needs to be rebased

**Comment by mmahadevan108** [comment]
I have rebased.

**Pull request merged**